src/SwedbankPay.Sdk/Identifiable.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+
+namespace SwedbankPay.Sdk
+{
+    /// <summary>
+    /// Contains a <seealso cref="Uri"/> to a uniquely identifiable resource.
+    /// </summary>
+    public class Identifiable : IIdentifiable
+    {
+        /// <summary>
+        /// Instantiates and sets the <see cref="Id"/> of the <see cref="Identifiable"/>.
+        /// </summary>
+        /// <param name="id">The unique ID of this resource.</param>
+        public Identifiable(Uri id)
+        {
+            Id = id;
+        }
+
+        /// <summary>
+        /// Instantiates and sets the <see cref="Id"/> of the <see cref="Identifiable"/>.
+        /// </summary>
+        /// <param name="id">The unique ID of this resource.</param>
+        internal Identifiable(string id)
+        {
+            if(string.IsNullOrEmpty(id) == false)
+            {
+                Id = new Uri(id, UriKind.RelativeOrAbsolute);
+            }

```

Comment:
If there's no URI available, the class inheriting `Identifiable` isn't actually identifiable and thus shouldn't inherit `Identifiable`. Constructing `Identifiable` such that the `Id` property is `null` is rather pointless.

```suggestion
            : this(new Uri(id, UriKind.RelativeOrAbsolute))
        {
```

Reply:
I guess. But not allowing `null` gives me 43 failed unit tests.